### ISSUE TITLE
[<port name>] build failure

### ISSUE THREAD

**Issue by gust138** [microsoft/vcpkg#40243]
### Operating system

Windows 10 

### Compiler

Cmake

### Steps to reproduce the behavior

```Shell
vcpkg install jsoncpp
```

### Failure logs

Computing installation plan...
The following packages will be built and installed:
    jsoncpp:x64-windows@1.9.5#4
Detecting compiler hash for triplet x64-windows...
Compiler found: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.40.33807/bin/Hostx64/x64/cl.exe
Restored 0 package(s) from C:\Users\Дарья\AppData\Local\vcpkg\archives in 102 us. Use --debug to see more details.
Installing 1/1 jsoncpp:x64-windows@1.9.5#4...
Building jsoncpp:x64-windows@1.9.5#4...
-- Downloading https://github.com/open-source-parsers/jsoncpp/archive/1.9.5.tar.gz -> open-source-parsers-jsoncpp-1.9.5.tar.gz...
[DEBUG] To include the environment variables in debug output, pass --debug-env
[DEBUG] Trying to load bundleconfig from C:\vcpkg\vcpkg-bundle.json
[DEBUG] Failed to open: C:\vcpkg\vcpkg-bundle.json
[DEBUG] Bundle config: readonly=false, usegitregistry=false, embeddedsha=nullopt, deployment=Git, vsversion=nullopt
[DEBUG] Metrics enabled.
[DEBUG] Feature flag 'binarycaching' unset
[DEBUG] Feature flag 'compilertracking' unset
[DEBUG] Feature flag 'registries' unset
[DEBUG] Feature flag 'versions' unset
[DEBUG] Feature flag 'dependencygraph' unset
warning: Download failed -- retrying after 1000ms
warning: Download failed -- retrying after 2000ms
warning: Download failed -- retrying after 4000ms
error: Missing open-source-parsers-jsoncpp-1.9.5.tar.gz and downloads are blocked by x-block-origin.
error: https://github.com/open-source-parsers/jsoncpp/archive/1.9.5.tar.gz: WinHttpSendRequest failed with exit code 12029
error: https://github.com/open-source-parsers/jsoncpp/archive/1.9.5.tar.gz: WinHttpSendRequest failed with exit code 12029
error: https://github.com/open-source-parsers/jsoncpp/archive/1.9.5.tar.gz: WinHttpSendRequest failed with exit code 12029
error: https://github.com/open-source-parsers/jsoncpp/archive/1.9.5.tar.gz: WinHttpSendRequest failed with exit code 12029
[DEBUG] D:\a\_work\1\s\src\vcpkg\base\downloads.cpp(1030):
[DEBUG] Time in subprocesses: 0us
[DEBUG] Time in parsing JSON: 4us
[DEBUG] Time in JSON reader: 0us
[DEBUG] Time in filesystem: 1136us
[DEBUG] Time in loading ports: 0us
[DEBUG] Exiting after 7.1 s (7084560us)

CMake Error at scripts/cmake/vcpkg_download_distfile.cmake:32 (message):

      Failed to download file with error: 1
      If you are using a proxy, please check your proxy setting. Possible causes are:

      1. You are actually using an HTTP proxy, but setting HTTPS_PROXY variable
         to `https://address:port`. This is not correct, because `https://` prefix
         claims the proxy is an HTTPS proxy, while your proxy (v2ray, shadowsocksr
         , etc..) is an HTTP proxy. Try setting `http://address:port` to both
         HTTP_PROXY and HTTPS_PROXY instead.

      2. If you are using Windows, vcpkg will automatically use your Windows IE Proxy Settings
         set by your proxy software. See https://github.com/microsoft/vcpkg-tool/pull/77
         The value set by your proxy might be wrong, or have same `https://` prefix issue.

      3. Your proxy's remote server is out of service.

      If you've tried directly download the link, and believe this is not a temporary
      download server failure, please submit an issue at https://github.com/Microsoft/vcpkg/issues
      to report this upstream download server failure.

Call Stack (most recent call first):
  scripts/cmake/vcpkg_download_distfile.cmake:270 (z_vcpkg_download_distfile_show_proxy_and_fail)
  scripts/cmake/vcpkg_from_github.cmake:106 (vcpkg_download_distfile)
  ports/jsoncpp/portfile.cmake:1 (vcpkg_from_github)
  scripts/ports.cmake:192 (include)

error: building jsoncpp:x64-windows failed with: BUILD_FAILEDSee https://learn.microsoft.com/vcpkg/troubleshoot/build-failures?WT.mc_id=vcpkg_inproduct_cli for more information.
Elapsed time to handle jsoncpp:x64-windows: 7.2 s
Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
  https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+jsoncpp
You can submit a new issue at:
  https://github.com/microsoft/vcpkg/issues/new?title=[jsoncpp]+Build+error+on+x64-windows&body=Copy+issue+body+from+C%3A%2Fvcpkg%2Finstalled%2Fvcpkg%2Fissue_body.md
  
[stdout-x64-windows.log](https://github.com/user-attachments/files/16478396/stdout-x64-windows.log)

**Comment by greenozon** [comment]
@gust138 
>WinHttpSendRequest failed with exit code 12029

looks like some temp issue with remote server that has the pkg archive
Please try again

I"ve tried now and it worked out of the box instantly, eg:

```
c:\Dev\vcpkg>vcpkg install jsoncpp
Computing installation plan...
The following packages will be built and installed:
    jsoncpp:x64-windows@1.9.5#4
Detecting compiler hash for triplet x64-windows...
Compiler found: C:/Dev/VS2019/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x64/cl.exe
Restored 0 package(s) from C:\Users\TATO\AppData\Local\vcpkg\archives in 419 us. Use --debug to see more details.
Installing 1/1 jsoncpp:x64-windows@1.9.5#4...
Building jsoncpp:x64-windows@1.9.5#4...
-- Downloading https://github.com/open-source-parsers/jsoncpp/archive/1.9.5.tar.gz -> open-source-parsers-jsoncpp-1.9.5.tar.gz...
-- Extracting source C:/Dev/vcpkg/downloads/open-source-parsers-jsoncpp-1.9.5.tar.gz
-- Using source at C:/Dev/vcpkg/buildtrees/jsoncpp/src/1.9.5-13b47286ba.clean
-- Configuring x64-windows
-- Building x64-windows-dbg
-- Building x64-windows-rel
-- Fixing pkgconfig file: C:/Dev/vcpkg/packages/jsoncpp_x64-windows/lib/pkgconfig/jsoncpp.pc
-- Using cached msys2-mingw-w64-x86_64-pkgconf-1~2.2.0-1-any.pkg.tar.zst.
-- Using cached msys2-msys2-runtime-3.5.3-3-x86_64.pkg.tar.zst.
-- Using msys root at C:/Dev/vcpkg/downloads/tools/msys2/1e74ca60daa10104
-- Fixing pkgconfig file: C:/Dev/vcpkg/packages/jsoncpp_x64-windows/debug/lib/pkgconfig/jsoncpp.pc
-- Installing: C:/Dev/vcpkg/packages/jsoncpp_x64-windows/share/jsoncpp/copyright
-- Performing post-build validation
Stored binaries in 1 destinations in 467 ms.
Elapsed time to handle jsoncpp:x64-windows: 9.9 s
jsoncpp:x64-windows package ABI: a10b406409ea40adea1bd455d5db6ae72a442363f5f93feed986edf59dbb371c
Total install time: 9.9 s
jsoncpp provides CMake targets:

  # this is heuristically generated, and may not be correct
  find_package(jsoncpp CONFIG REQUIRED)
  target_link_libraries(main PRIVATE JsonCpp::JsonCpp)

jsoncpp provides pkg-config modules:

  # A C++ library for interacting with JSON
  jsoncpp


c:\Dev\vcpkg>vcpkg install jsoncpp
Computing installation plan...
The following packages are already installed:
    jsoncpp:x64-windows@1.9.5#4
jsoncpp:x64-windows is already installed
Total install time: 705 us
jsoncpp provides CMake targets:

  # this is heuristically generated, and may not be correct
  find_package(jsoncpp CONFIG REQUIRED)
  target_link_libraries(main PRIVATE JsonCpp::JsonCpp)

jsoncpp provides pkg-config modules:

  # A C++ library for interacting with JSON
  jsoncpp
```

**Comment by greenozon** [comment]
another reason maybe this one:
>WinHttpSendRequest failed with exit code 12029

This problem occurs because Windows is not configured to use TLS 1.2 or 1.3 protocol and the hosted site has disabled support for TLS 1.0 and 1.1 in their on-premises environment (TLS 1.0 and 1.1 are nearing end-of-life, not to mention a security risk and many hosting sites have started disabling/phasing it out).
